devito/passes/iet/misc.py

```diff
@@ -196,42 +199,50 @@ def _generate_macros_findexeds(iet, sregistry=None, tracker=None, **kwargs):
     return iet
 
 
-def _generate_macros_math(iet):
+def _generate_macros_math(iet, langbb=None):
     headers = []
+    includes = []
     for i in FindApplications().visit(iet):
-        headers.extend(_lower_macro_math(i))
+        header, include = _lower_macro_math(i, langbb)
+        headers.extend(header)
+        includes.extend(include)
 
-    return headers
+    return headers, set(includes) - {None}
 
 
 @singledispatch
-def _lower_macro_math(expr):
-    return ()
+def _lower_macro_math(expr, langbb):
+    return (), {}
 
 
 @_lower_macro_math.register(Min)
 @_lower_macro_math.register(sympy.Min)
-def _(expr):
-    if has_integer_args(*expr.args) and len(expr.args) == 2:
-        return (('MIN(a,b)', ('(((a) < (b)) ? (a) : (b))')),)
+def _(expr, langbb):
+    if has_integer_args(*expr.args):
+        return (('MIN(a,b)', ('(((a) < (b)) ? (a) : (b))')),), {}
     else:
-        return ()
+        return (), as_tuple(langbb.get('header-algorithm'))
```

Comment:
u see for example here how this should really be called `header-math`, as `algorithm` is kinda misleading since it's all but obvious to think immediately about CXX's algorithm lib